src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -345,6 +356,79 @@ def _list_bids_aggregator(self, arguments):
                 "transaction_id": arguments.get("transaction_id", None)}
 
 
+class LoadForecastExternalStrategy(LoadExternalMixin, LoadForecastStrategy):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def channel_dict(self):
+        return {**self._channel_dict,
+                f'{self.channel_prefix}/set_load_power_forecast': self._set_power_forecast}
```

Comment:
Since all strategies have unique channel prefix, we can  generalise the suffix like here `set_load_power_forecast` to simply `power_forecast`

Reply:
Agreed!

Reply:
@faizan2590 Sorry, I didn't read properly the first time.
I thought about this when implementing too, and then the `_set_power_forecast` could even live in the ExternalMixin.
However, in the `_set_power_forecast_impl` different Strategy attributes are set and this shows also that pv and load are not identical here. This is why I kept the difference between load and pv.
But if @spyrostz agrees, I will refactor it as you said.

Reply:
@faizan2590 Actually, no, i think yours is a goof idea and will try to implement it like this.

Reply:
> I thought about this when implementing too, and then the _set_power_forecast could even live in the ExternalMixin.

As per MRO `_set_power_forecast ` would be overwritten by `LoadForecastExternalStrategy` and `PVForecastExternalStrategy` respectively, so I don't  think you should be worried about base implementation in `ExternalMixin `

Reply:
Og course this includes to remove the implementations in `LoadForecastExternalStrategy` and `PVForecastExternalStrategy`, currently trying it.